common/connection.go

```diff
@@ -321,13 +319,7 @@ func (c *Connection) findTargetIDForLog(id target.SessionID) target.ID {
 }
 
 func (c *Connection) recvLoop() {
-	// Setting the wsBuffer as the same size as bufio.NewReader. We should look at
-	// optimizing it depending on the average website, or allow it to be
-	// configurable, but it depends on how much this affects the user.
-	wsBuffer := make([]byte, 4096)
-	// This buffer grows as the test progresses.
-	var recvBuffer []byte
-	var bufReader bufio.Reader
+	var buffer bytes.Buffer
```

Comment:
> It's nice that we find a solution for this tedious issue 🎉
> 
> I have suggestions.
> 
> This PR adds this:
> 
> https://github.com/grafana/xk6-browser/blob/66179668cb8bc691b1150931dd2f56161e12df36/common/connection.go#L324-L362
> 
> I believe keeping this already tricky and central part of the code simpler will serve us better in the long run.
> 
> Also, correctly using `io.Reader` directly is risky and highly likely to create unforeseen issues. Since our goal is to reuse the same buffer instead of allocating a new one, we can use `bytes.Buffer` to let it handle the `io.Reader`. It reuses the same internal bytes buffer. Go has `bytes.Buffer` for these kind of situations:
> 
> ```go
> var buffer bytes.Buffer
> 
> // loop:
> if _, err := buffer.ReadFrom(reader); err != nil {
>     // ...
> }
> 
> // Clone so that we can share the underlying bytes
> buf := bytes.Clone(buffer.Bytes())
> 
> // Reuse the underlying byte slice
> buffer.Reset()
> 
> // Use buf here
> ...
> ```

I ran this suggested change through my test and it all checks off Perfectly 👍 

> If your profiling shows that `bufio.Reader` is helpful, you can wrap `reader` with it: `buffer.ReadFrom(bufio.NewReader(reader))`.

This didn't make a difference, so I've left it out.

> 💡 It also makes sense to add a `Benchmark` to see how much memory usage we reduce and keep it at that level.

I've added a task in the original [issue](https://github.com/grafana/xk6-browser/issues/1480) tracking all the memory related issues. I can't create a benchmark test yet as they aren't behaving as we would expect since the memory keeps going up due to the other memory leaks (investigating in https://github.com/grafana/xk6-browser/pull/1484). Once memory leaking/usage has come under a bit more control then I think it would be best to complete the benchmarks.

Reply:
> I ran this suggested change through my test and it all checks off Perfectly 👍 https://github.com/grafana/xk6-browser/commit/a631b38457c5ff4497ae5292cc3e7735ee230c8e

Nice 🎉 

> I've added a task in the original https://github.com/grafana/xk6-browser/issues/1480 tracking all the memory related issues. I can't create a benchmark test yet as they aren't behaving as we would expect since the memory keeps going up due to the other memory leaks (investigating in https://github.com/grafana/xk6-browser/pull/1484). Once memory leaking/usage has come under a bit more control then I think it would be best to complete the benchmarks.

This is fine, but benchmarks can also help with pprofing.